### PR TITLE
add very basic tests of win_package, based on existing win_msi tests.

### PR DESCRIPTION
**Pull request by jhawkesworth**
##### ISSUE TYPE

- Feature Pull Request
##### COMPONENT NAME

win_package
##### ANSIBLE VERSION

```
ansible 2.2.0 (test_win_package 519815f223) last updated 2016/09/04 06:53:46 (GMT +100)
  lib/ansible/modules/core: (detached HEAD 6eab2b3d40) last updated 2016/08/28 10:52:27 (GMT +100)
  lib/ansible/modules/extras: (detached HEAD 628ee6864d) last updated 2016/09/04 06:24:26 (GMT +100)
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides

```
##### SUMMARY

As discussed at modules review a few weeks ago, win_msi is going to be deprecated, so this pr adds some basic tests for win_package, which will be the prefered way of installing msi (and .exe packaged windows installers) in future.

Doesn't fix a specific issue, but there's plenty of examples of people struggling with win_msi on the ansible project google group.

```
before is not applicable (new tests)

after:
export TEST_FLAGS='-i /etc/ansible/hosts -t test_win_package -l 12'
time make test_win_group2
rm -rf ~/ansible_testing
mkdir -p ~/ansible_testing
ansible-playbook test_win_group2.yml -i inventory.winrm -e outputdir=~/ansible_testing -e @integration_config.yml  -v -i /etc/ansible/hosts -t test_win_package -l 12
Using /etc/ansible/ansible.cfg as config file

PLAY [windows] *****************************************************************

TASK [test_win_package : use win_get_url module to download msi] ***************
changed: [WIN-12] => {"changed": true, "win_get_url": {"dest": "C:\\Program Files\\7z922-x64.msi", "url": "https://ansible-ci-files.s3.amazonaws.com/test/integration/roles/test_win_msi/7z922-x64.msi"}}

TASK [test_win_package : make sure msi is uninstalled] *************************
 [WARNING]: SyntaxError in safe_eval() on expr:
{23170F69-40C1-2702-0922-000001000000} (invalid syntax (<unknown>, line 1))

ok: [WIN-12] => {"changed": false, "name": "C:\\Program Files\\7z922-x64.msi"}

TASK [test_win_package : install msi] ******************************************
 [WARNING]: SyntaxError in safe_eval() on expr:
{23170F69-40C1-2702-0922-000001000000} (invalid syntax (<unknown>, line 1))

changed: [WIN-12] => {"changed": true, "name": "C:\\Program Files\\7z922-x64.msi"}

TASK [test_win_package : check win_package install result] *********************
ok: [WIN-12] => {"changed": false, "msg": "all assertions passed"}

TASK [test_win_package : install msi again with argument] **********************
 [WARNING]: SyntaxError in safe_eval() on expr:
{23170F69-40C1-2702-0922-000001000000} (invalid syntax (<unknown>, line 1))

ok: [WIN-12] => {"changed": false, "name": "C:\\Program Files\\7z922-x64.msi"}

TASK [test_win_package : check win_package install again result] ***************
ok: [WIN-12] => {"changed": false, "msg": "all assertions passed"}

TASK [test_win_package : uninstall msi] ****************************************
 [WARNING]: SyntaxError in safe_eval() on expr:
{23170F69-40C1-2702-0922-000001000000} (invalid syntax (<unknown>, line 1))

changed: [WIN-12] => {"changed": true, "name": "C:\\Program Files\\7z922-x64.msi"}

TASK [test_win_package : check win_package uninstall result] *******************
ok: [WIN-12] => {"changed": false, "msg": "all assertions passed"}

PLAY RECAP *********************************************************************
WIN-12                     : ok=8    changed=3    unreachable=0    failed=0   


real    0m15.097s
user    0m2.436s
sys 0m0.328s

```

### CODE REVIEW

**Comment by jhawkesworth** [comment]
as promised @gundalow , @nitzmahone

File: test/integration/roles/test_win_package/defaults/main.yml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+
+msi_url: https://ansible-ci-files.s3.amazonaws.com/test/integration/roles/test_win_msi/7z922-x64.msi
```
Comment (gundalow) [inline test/integration/roles/test_win_package/defaults/main.yml:3]: Who controls `ansible-ci-files.s3.amazonaws.com`?

File: test/integration/roles/test_win_package/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    url: "{{msi_url}}"
+    dest: "{{msi_download_path}}"
+  register: win_get_url_result
+
+- name: make sure msi is uninstalled
+  win_package:
+    path: "{{msi_download_path}}"
+    product_id: "{{msi_product_code}}"
+    state: absent
+#  ignore_errors: true
```
Comment (gundalow) [inline test/integration/roles/test_win_package/tasks/main.yml]: Was `ignore_errors` left in from development, does that need removing now?

**Comment by gundalow** [comment]
Looks good. It would be good to have some idempotent tests, e.g.
install -> check it's changed
install again (same arguments) -> check it's unchanged

File: test/integration/roles/test_win_package/defaults/main.yml (hunk 1 above)
Comment (jhawkesworth) [reply test/integration/roles/test_win_package/defaults/main.yml:3]: hmm, that must have changed recently (probably just to reduce likelihood of network failures, as I think it used to come from sourceforge, or wherever the 7 zip project is).  my guess would be @mattclay ?

File: test/integration/roles/test_win_package/tasks/main.yml (hunk 2 above)
Comment (jhawkesworth) [reply test/integration/roles/test_win_package/tasks/main.yml]: that's a left over from the win_msi tests, which I think were unreliable.  I agree its confusing as it is though so could be deleted.

**Comment by jhawkesworth** [comment]
thanks for the review @gundalow - I agree an idempotency test would be good to add - wanted to get something in before freeze if possible, because of the plan afoot to deprecate win_msi in 2.2.

**Comment by gundalow** [comment]
Could you please add an idempotency check then we can get this merged?

**Comment by jhawkesworth** [comment]
Added idempotency check for state=absent case (state=present) was already there but badly named, so I renamed that too, and removed pointless commented out ignore_errors: true.

Hoping this can get in before the freeze.

**Comment by gundalow** [comment]
Thanks :)

**Comment by jhawkesworth** [comment]
Great, thanks for the speedy merge!

**Comment by nitzmahone** [comment]
For future reference, tests are generally not subject to the freeze (just
product features)- we'll ALWAYS take good tests. :)

**Comment by jhawkesworth** [comment]
thanks, worth knowing.  I've got a couple of other things I want to finish up but I'd like to come back round to this and add some more tests.

File: test/integration/roles/test_win_package/defaults/main.yml (hunk 1 above)
Comment (mattclay) [reply test/integration/roles/test_win_package/defaults/main.yml:3]: That s3 bucket is in an Ansible AWS account. It was added as a more reliable place to put binaries for tests without having to put them in the git repo.

**Pull request merged**